CMakeLists.txt

```diff
@@ -63,6 +64,9 @@ if(ENABLE_SANITIZER)
   set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -fsanitize=address")
 endif()
 
+if(USE_STD_REGEX)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DUSE_STD_REGEX=1")
```

Comment:
better would be to use target properties and set the compiler flags at 
https://github.com/valhalla/valhalla/blob/b860d71c2581ff9a5dd7af97b03e599ec7ba4274/src/CMakeLists.txt#L77
as 
```
  target_compile_definitions(${library}
    PUBLIC
      RAPIDJSON_HAS_STDSTRING
    PRIVATE
      $<$<BOOL:${USE_STD_REGEX}>:USE_STD_REGEX=1>)
```

and do not modify global `CMAKE_CXX_FLAGS` variable